Modern/ops/src/main/java/org/bonej/ops/mil/MILPlane.java

```diff
@@ -0,0 +1,294 @@
+ * filled, and a part of a larger whole, e.g. a section of trabecular bone.
+ * </p>
+ * <p>
+ * For more details, see:
+ * </p>
+ * <ul>
+ * <li>Moreno et al., <a href=
+ * "http://liu.diva-portal.org/smash/get/diva2:533443/FULLTEXT01.pdf">On the
+ * Efficiency of the Mean Intercept Length Tensor</a></li>
+ * <li>Odgaard A (1997) Three-dimensional methods for quantification of
+ * cancellous bone architecture. Bone 20: 315-28. <a href=
+ * "http://dx.doi.org/10.1016/S8756-3282(97)00007-0">doi:10.1016/S8756-3282(97)00007-0</a>.</li>
+ * <li>Harrigan TP, Mann RW (1984) Characterization of microstructural
+ * anisotropy in orthotropic materials using a second rank tensor. J Mater Sci
+ * 19: 761-767. <a href=
+ * "http://dx.doi.org/10.1007/BF00540446">doi:10.1007/BF00540446</a>.</li>
+ * </ul>
+ *
+ * @author Richard Domander
+ */
+@Plugin(type = Op.class)
+public class MILPlane<B extends BooleanType<B>> extends
+	AbstractBinaryFunctionOp<RandomAccessibleInterval<B>, AxisAngle4d, Vector3d>
+	implements Contingent
+{
+
+	private static BinaryHybridCFI1<Tuple3d, AxisAngle4d, Tuple3d> rotateOp;
+	private static BinaryFunctionOp<ValuePair<Point3d, Vector3d>, Interval, Optional<ValuePair<DoubleType, DoubleType>>> intersectOp;
+	private final Random random = new Random();
+	/**
+	 * Number of sampling lines generated per dimension.
+	 * <p>
+	 * Each bin represents a sub-region of the plane where a sampling line
+	 * originates. For example, if bins = 2 then we get four lines originating
+	 * from the four quadrants of the plane.
+	 * </p>
+	 * <p>
+	 * If left null, bins is set to <em>d</em>, where <em>d</em> is the largest
+	 * diagonal of the input interval.
+	 * </p>
+	 * <p>
+	 * The higher the number, the more likely you get an accurate result.
+	 * </p>
+	 */
+	@Parameter(required = false, persist = false)
+	private Long bins;
+	/**
+	 * The scalar step between positions on a sampling line where voxels are read.
+	 * <p>
+	 * For example, an increment of 1.0 adds a vector of length 1.0 to the
+	 * position.
+	 * </p>
+	 * <p>
+	 * If left null, the increment is set to 1.0.
+	 * </p>
+	 * <p>
+	 * The higher the number, the more likely you get an accurate result.
+	 * </p>
+	 */
+	@Parameter(required = false, persist = false)
+	private Double increment;
+	/**
+	 * The seed used in the random generator in the algorithm.
+	 * <p>
+	 * Given that all other parameters all the same (including the interval and
+	 * rotation), then using the same seed will give the same results.
+	 * </p>
+	 * <p>
+	 * If left null, a new generator is created with the default constructor.
+	 * </p>
+	 */
+	@Parameter(required = false, persist = false)
+	private Long seed;
+
+	/**
+	 * Calculates the MIL vector of the interval.
+	 *
+	 * @param interval a 3D interval.
+	 * @param rotation direction of the MIL lines in the interval.
+	 * @return total length / total phase changes of vectors sampled in interval.
```

Comment:
detail: This could be slightly more precise: possibly "a vector with magnitude total length sampled / total phase changes from background to foreground and direction of the MIL lines"